pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -322,10 +333,45 @@ func (ctrl *PersistentVolumeController) emitEventForUnboundDelayBindingClaim(cla
 			message = fmt.Sprintf("waiting for pod %s to be scheduled", podNames[0])
 		}
 	}
+	// we already emitted an event and we have the condition set, do nothing
+	if c := findStatusCondition(claim.Status.Conditions, v1.PersistentVolumeClaimWaitingForConsumer); c != nil && c.Status == v1.ConditionTrue {
+		return nil
+	}
+
+	// this is the first time we see waiting for consumer, set the condition and emit event
+	newClaim := claim.DeepCopy()
+	newClaim.Status.Conditions = append(newClaim.Status.Conditions, v1.PersistentVolumeClaimCondition{
+		Type:               v1.PersistentVolumeClaimWaitingForConsumer,
+		Status:             v1.ConditionTrue,
+		LastTransitionTime: metav1.Now(),
+		Reason:             "WaitingForConsumer",
+		Message:            message,
+	})
+
+	if _, err := ctrl.GetKubeClient().CoreV1().PersistentVolumeClaims(claim.GetNamespace()).Update(context.TODO(), newClaim, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
 	ctrl.eventRecorder.Event(claim, v1.EventTypeNormal, reason, message)
 	return nil
 }
 
+// cleanWaitingForConsumerCondition remove the WaitingForConsumer status condition if set.
+func (ctrl *PersistentVolumeController) cleanWaitingForConsumerCondition(claim *v1.PersistentVolumeClaim) (bool, error) {
+	if c := findStatusCondition(claim.Status.Conditions, v1.PersistentVolumeClaimWaitingForConsumer); c != nil && c.Status == v1.ConditionTrue {
+		newClaim := claim.DeepCopy()
+		newClaim.Status.Conditions = []v1.PersistentVolumeClaimCondition{}
+		for _, c := range claim.Status.Conditions {
+			if c.Type == v1.PersistentVolumeClaimWaitingForConsumer {
+				continue
+			}
+			newClaim.Status.Conditions = append(newClaim.Status.Conditions, c)
+		}
+		_, err := ctrl.GetKubeClient().CoreV1().PersistentVolumeClaims(claim.GetNamespace()).Update(context.TODO(), newClaim, metav1.UpdateOptions{})
```

Comment:
this is updating a condition, so shouldn't this be calling `UpdateStatus`? does that mean that this is not actually working as-is? if so, what tests should be added to exercise this?